cmake/FindFreeRTOS.cmake

```diff
@@ -1,117 +1,195 @@
+set(FreeRTOS_PORTS ARM_CM0 ARM_CM3 ARM_CM4F ARM_CM7)
```

Comment:
Missing ports to test:
CM4 exists (STM32F2)
CM0+ is working ? (STM32L0)
CM33/CM33F (STM32L5)

Reply:
I have added the MPU versions of CM3 and CM4. I have not added CM33 yet as it has another source code structure (`secure` and `non-secure` folder) which would require special CMake handoing, and this would be better as a separate PR.

I have not tested the MPU versions yet, I don't really have hardware for it but I can try compiling it

Reply:
#231